binding/SkiaSharp.NativeAssets.WebAssembly/buildTransitive/SkiaSharp.targets

```diff
@@ -14,15 +19,25 @@
   </ItemGroup>
 
   <!-- If this is a ASP.NET Blazor web assembly app -->
-  <PropertyGroup Condition="'$(UsingMicrosoftNETSdkBlazorWebAssembly)' == 'true'">
+  <PropertyGroup Condition="'$(UsingMicrosoftNETSdkBlazorWebAssembly)' == 'true' and '$(ShouldIncludeNativeSkiaSharp)' == 'True'">
     <WasmBuildNative Condition="'$(WasmBuildNative)' == ''">true</WasmBuildNative>
     <EmccExtraLDFlags>$(EmccExtraLDFlags) -s USE_WEBGL2=1</EmccExtraLDFlags>
+    <!-- Pick the threading type: 'mt' for multi-threading or 'st' for single-threading -->
+    <_SkiaSharpNativeBinaryType Condition="'$(WasmEnableThreads)' == 'True'">mt</_SkiaSharpNativeBinaryType>
+    <_SkiaSharpNativeBinaryType Condition="'$(WasmEnableThreads)' != 'True'">st</_SkiaSharpNativeBinaryType>
+    <!-- Pick the SIMD support: 'simd' for supported or '' for not-supported -->
+    <_SkiaSharpNativeBinaryType Condition="'$(WasmEnableSIMD)' == 'True'">$(_SkiaSharpNativeBinaryType),simd</_SkiaSharpNativeBinaryType>
+    <_SkiaSharpNativeBinaryType Condition="'$(WasmEnableSIMD)' != 'True'">$(_SkiaSharpNativeBinaryType)</_SkiaSharpNativeBinaryType>
   </PropertyGroup>
-  <ItemGroup Condition="'$(UsingMicrosoftNETSdkBlazorWebAssembly)' == 'true' and '$(TargetFrameworkVersion)' != ''">
+  <ItemGroup Condition="'$(UsingMicrosoftNETSdkBlazorWebAssembly)' == 'true' and '$(TargetFrameworkVersion)' != '' and '$(ShouldIncludeNativeSkiaSharp)' == 'True'">
```

Comment:
Since `NativeFileReference` format is consistent for any Mono WASM project, would it make sense to use another property here?

Imagine something like:
```xml
<PropertyGroup>
    <ShouldIncludeNativeSkiaSharp Condition="'$(ShouldIncludeNativeSkiaSharp)' == '' AND '$(UsingMicrosoftNETSdkBlazorWebAssembly)' == 'true'">true</ShouldIncludeNativeSkiaSharp>
</PropertyGroup>
```

This way third party, like Avalonia, can set `ShouldIncludeNativeSkiaSharp` to true.
And it will still support Blazor as a first party by defaulting to `UsingMicrosoftNETSdkBlazorWebAssembly` value.

Reply:
Minor note, Native AOT uses `<NativeLibrary />` [instead of](https://github.com/AvaloniaUI/Avalonia/commit/4134420b715a613ca5bc89a7ef47a0236f5540f0#diff-a4331dea622d07577e3da0835bbc5225c5554f2be6b43557beb0538205af4194R26-R28) `<NativeFileReference/>`. But that's likely wouldn't be a concern here until .NET 10 or 11.

Reply:
@jeromelaban you folks have a different way of loading native things? You just read the `@Content` and figure it out. Are there some checks that if we add will cause issues?

@maxkatz6 are you just the normal logic and no special checks? So you are suggesting that we set it up such that `ShouldIncludeNativeSkiaSharp =True` always includes the bits without doing fancy checks, and leave the blazor check for rather the default value of `ShouldIncludeNativeSkiaSharp`?

And for Uno, this means that it will not be set by default but we use the `$(IsUnoHead)` value to add the Uno way. 

@maxkatz6 is there a `Is Avalonia App` property that I can check to always include? OR would you rather it be opt in?

Reply:
@jeromelaban you folks have a different way of loading native things? You just read the @(Content) and figure it out. Are there some checks that if we add will cause issues?

Indeed, this predates the support from the runtime, but it also supports automatic selection based on the path's format.  Starting from net9, we'll transitively include the `NativeFileReference` support.

> And for Uno, this means that it will not be set by default but we use the $(IsUnoHead) value to add the Uno way.

We could also add support that property, but in general, the fact that `'$(OutputType)' == 'exe'` may be enough to conditionally include assets. We had that `$(IsUnoHead)` property in the past because some targets were requiring the head to be a `dll`, not an `exe`.

As for the condition suggested by @maxkatz6, I'd adjust it as

Reply:
@jeromelaban you folks have a different way of loading native things? You just read the @(Content) and figure it out. Are there some checks that if we add will cause issues?

Indeed, this predates the support from the runtime, but it also supports automatic selection based on the path's format.  Starting from net9, we'll transitively include the `NativeFileReference` support.

> And for Uno, this means that it will not be set by default but we use the $(IsUnoHead) value to add the Uno way.

We could also add support that property, but in general, the fact that `'$(OutputType)' == 'exe'` may be enough to conditionally include assets. We had that `$(IsUnoHead)` property in the past because some targets were requiring the head to be a `dll`, not an `exe`.

As for the condition suggested by @maxkatz6, I'd adjust it as

Reply:
@jeromelaban you folks have a different way of loading native things? You just read the @(Content) and figure it out. Are there some checks that if we add will cause issues?

Indeed, this predates the support from the runtime, but it also supports automatic selection based on the path's format.  Starting from net9, we'll transitively include the `NativeFileReference` support.

> And for Uno, this means that it will not be set by default but we use the $(IsUnoHead) value to add the Uno way.

We could also add support that property, but in general, the fact that `'$(OutputType)' == 'exe'` may be enough to conditionally include assets. We had that `$(IsUnoHead)` property in the past because some targets were requiring the head to be a `dll`, not an `exe`.

As for the condition suggested by @maxkatz6, I'd adjust it as:

```xml
<PropertyGroup> <ShouldIncludeNativeSkiaSharp Condition="'$(ShouldIncludeNativeSkiaSharp)' == '' AND ( '$(UsingMicrosoftNETSdkBlazorWebAssembly)' == 'true' OR '$(UsingMicrosoftNETSdkWebAssembly)' == 'true' )">true</ShouldIncludeNativeSkiaSharp> </PropertyGroup>
```

to support the non-blazor Wasm SDK.

Reply:
> @maxkatz6 is there a Is Avalonia App property that I can check to always include?

No, not really.
For our needs, there is a little of advantage of adding avalonia specific checks into SkiaSharp, since we completely reuse runtime NativeFileReference support. I.e. without any special handling.

The only exception of special handling was NativeAOT, but even then, it's the same build-in runtime feature (just a different runtime), not something Avalonia specific.

Reply:
> As for the condition suggested by @maxkatz6, I'd adjust it as:

Yes, this adjustment looks good. It will automatically work with Avalonia too, as our templates use Sdk.WebAssembly by default.
Manually settable flag like `ShouldIncludeNativeSkiaSharp` is still desired, as `Sdk.WebAssembly` itself is optional, and some older Avalonia apps still run without it.

Reply:
I updated the logic to still keep the ShouldIncludeNativeSkiaSharp as a flag that is always true. Then I did what @jeromelaban suggested by adding both the sdk properties to the main inclusion. I don't want to make the ShouldIncludeNativeSkiaSharp have any other default as if we need to do something later on, then we may be stuck. However, the eventual logic is the same.

@maxkatz6 @jeromelaban  will the new logic work, or am I missing something that you are seeing?

Reply:
@mattleibow these changes look good to me. 

The only thing, native files inclusion can't be forced, unless user uses correct SDK. As WASM projects can be used with a standard SDK just fine. But this change improves defaults anyway.

Reply:
When you say forced, are you saying I am forcing something here? Or are you saying it as a precaution?

Reply:
No, no.
I mean, some developer might want to enable ShouldIncludeNativeSkiaSharp, even if they don't use WASM SDK. Since it's possible to run Browser apps without WASM SDK. Most of older Avalonia Browser apps don't use WASM SDK for example.

It's not possible to force this property right now. But is possible to manually include these files, so that's not a blocker.

Reply:
By WASM SDK I mean `<Project SDK="Microsoft.NET.Sdk.WebAssembly">` specifically. This PR includes a check for this (or Blazor) SDK.

Reply:
Oh, I see. The opposite of what I was thinking. You _want_to be able to force the native. Did you use the same item group?